### PR TITLE
refactor: batch slot management in decoder classes

### PR DESCRIPTION
**Pull request by Funatiq**
Highlights:
* Set up batch slots in `MakeDecodingBatchInputOutput` (outside of the decoder).
* Set up logits vector in `MakeDecodingBatchInputOutput` (outside of the decoder).
* Remove `active` vector from `decoder_batch::Input`.
* Unify the creation of decoder batch inputs in algorithm and tests.

See commit messages for details.

### CODE REVIEW

**Comment by Funatiq** [comment]
/bot run --stage-list "H100_PCIe-TensorRT-1, H100_PCIe-TensorRT-3"

File: cpp/include/tensorrt_llm/batch_manager/decoderBuffers.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ class DecoderInputBuffers
     TensorPtr forwardBatchSlotsRequestOrderDevice;
     TensorPtr fillValues;
     TensorPtr fillValuesDevice;
-    TensorPtr forwardBatchSlots;
+    std::vector<TensorPtr> forwardBatchSlots;
```
Comment (dcampora) [inline cpp/include/tensorrt_llm/batch_manager/decoderBuffers.h:51]: Why this change?

**Review by dcampora — APPROVED** [review]

File: cpp/include/tensorrt_llm/batch_manager/decoderBuffers.h (hunk 1 above)
Comment (Funatiq) [reply cpp/include/tensorrt_llm/batch_manager/decoderBuffers.h:51]: So each tensor can be resized to the actual batch size. Previously it was a sparse 2-dim tensor. Also see [here](https://github.com/NVIDIA/TensorRT-LLM/pull/3300/files#diff-a86768fcf5826f655591d71de3615f66d0f5b55583c7329b6007cfa21c6f9e1fR73).
